integration-app/src/main/java/uk/gov/crowncommercial/esourcing/auth/AuthConfiguration.java

```diff
@@ -14,17 +15,20 @@
 @EnableWebSecurity
 public class AuthConfiguration extends WebSecurityConfigurerAdapter {
 
-  // TODO (pillingworth, 2020-01-08) make this configurable in the application.properties
-  private static final String API_KEY_AUTH_HEADER_NAME = "x-api-key";
-
-  @Value("${ccs.esourcing.tenders.apikey:}")
+  @Value("${ccs.esourcing.tenders.apikeyheader:x-api-key}")
+  private String apiKeyHeader;
+  
+  @Value("${ccs.esourcing.tenders.apikeys:}")
   private Set<String> apiKeys;
+  
+  @Value("${ccs.esourcing.tenders.ipallowlist:}")
+  private Set<String> ipAllowList;
 
   @Override
   protected void configure(HttpSecurity http) throws Exception {
 
-    ApiKeyAuthFilter apiKeyFilter = new ApiKeyAuthFilter(API_KEY_AUTH_HEADER_NAME);
-    apiKeyFilter.setAuthenticationManager(new ApiKeyAuthManager(apiKeys));
+    ApiKeyAuthFilter apiKeyFilter = new ApiKeyAuthFilter(apiKeyHeader);
+    apiKeyFilter.setAuthenticationManager(apiKeyAuthManager());
 
     // @formatter:off
     http.addFilterBefore(ipAddressFilter(), AbstractPreAuthenticatedProcessingFilter.class)
```

Comment:
Do we still need antMatchers("/actuator/**").permitAll() if you've added it to the filter?

and/or

Should we include openapi.yaml & favicon.ico in the excludePaths?

Reply:
So, the request is received by the IpAddressFilter first before anything else. It checks the IP addresses and only lets through calls for those that are in the allow list or if the path is excluded. We only exclude the actuator so the health check works from an internal IP address. We could put an IP address configured for 10.0.0.x for these too if we wanted to tie it all down. This isn't authentication - just simple request filtering.

The ANT filter controls what requires authentication. Everything requires auth except the actuator, and the openapi.yaml and favicon.ico. 

What I have I think is right.